jupyterlab/labextensions.py

```diff
@@ -304,6 +330,12 @@ class UpdateLabExtensionApp(BaseExtensionApp):
     all = Bool(False, config=True, help="Whether to update all extensions")
 
     def run_task(self):
+        self.deprecation_warning(
+            """Updating extensions with the jupyter labextension update command is now deprecated and will be removed in a future major version of JupyterLab.
+
+Users should manage prebuilt extensions with package managers like pip and conda, and extension authors are encouraged to distribute their extensions as Python packages.
```

Comment:
```suggestion
Users should manage prebuilt extensions with package managers like pip and conda, and extension authors are encouraged to distribute their extensions as prebuilt packages.
```

I would try to avoid "Python extensions" phrase as this gives an incorrect impression that "JupyterLab only supports Python". For example IHaskell distributes their prebuilt extensions without any Python at all:  https://github.com/IHaskell/IHaskell/issues/1238#issuecomment-907658217, which was the intent behind having the extensions system support arbitrary source of prebuilt packages.

This also applies to the same part below and and above. For consistency, the repetitive part could possible be moved to a constant variable to make adjustments easier.

Reply:
Thanks for the suggestion. I think "Python" was used here because the new extension manager in 4.0 now shows up as "PyPI manager" by default.

Agree we can switch to prebuilt and make some of these strings easier to update later on :+1: